src/main/java/com/ratifire/devrate/service/NotificationService.java

```diff
@@ -150,65 +150,57 @@ private Notification createNotification(NotificationType type, NotificationPaylo
   private void addNotification(Notification notification, User user) {
     notificationRepository.save(notification);
 
-    sendUserNotifications(user.getId());
+    sendUserNotification(user.getId(), mapper.toDto(notification));
   }
 
   /**
    * Deletes a notification by its ID and send updated notifications.
    *
-   * @param userId         The ID of the user associated with the notification.
    * @param notificationId The ID of the notification to delete.
    */
-  public void deleteById(long userId, long notificationId) {
+  public void deleteById(long notificationId) {
     notificationRepository.deleteById(notificationId);
-
-    sendUserNotifications(userId);
   }
 
   /**
-   * Sends the notifications to the user via WebSocket.
+   * Sends a notification to the specified user via WebSocket.
    *
-   * @param userId The ID of the user to send updated notifications for.
+   * @param userId       The ID of the user to send fresh notification for.
+   * @param notification The notification data to be sent.
    */
-  private void sendUserNotifications(long userId) {
+  private void sendUserNotification(long userId, NotificationDto notification) {
     UserSecurity userSecurity = userSecurityService.getByUserId(userId);
```

Comment:
what do you say about replacing the key from email to Long userId in Map? so we can avoid unnecessary request to the database